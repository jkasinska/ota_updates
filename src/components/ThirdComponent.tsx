import React from 'react';
import { SafeAreaView, View, VirtualizedList, StyleSheet, Text } from 'react-native';

interface Item {
  key: number,
  id: string,
  title: string
}

const DATA: Item[] = [];

const getItem = (data: Item[], index: number) => {
  return {
    key: index.toString(),
    id: Math.random().toPrecision(2).toString(),
    title: `Item ${index+1}`
  }
}

const getItemCount = () => {
  return 200;
}

const ThirdComponent = () => {
  return (
    <SafeAreaView style={styles.container}>
      <VirtualizedList
        data={DATA}
        initialNumToRender={4}
        renderItem={({ item }) => {
          return (
            <View style={styles.item}>
              <Text style={styles.title}>{item.title}</Text>
            </View>
          )
        }}
        keyExtractor={item => item.key}
        getItemCount={getItemCount}
        getItem={getItem}
      />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 20,
  },
  item: {
    backgroundColor: '#f9c2ff',
    height: 150,
    justifyContent: 'center',
    marginVertical: 8,
    marginHorizontal: 16,
    padding: 20,
  },
  title: {
    fontSize: 32,
  },
});

export default ThirdComponent;
