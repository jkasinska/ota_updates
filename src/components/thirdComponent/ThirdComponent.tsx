import React from 'react';
import {
  SafeAreaView,
  View,
  VirtualizedList,
  StyleSheet,
  Text,
  Button,
} from 'react-native';
import { NavigationProp, RouteProp } from '@react-navigation/native';

interface ThirdComponentProps {
  navigation: NavigationProp<any>;
  route: RouteProp<any, any>;
}

type Item = {
  key: number;
  id: string;
  title: string;
};

const DATA: Item[] = [];

const getItem = (data: Item[], index: number) => {
  return {
    key: index.toString(),
    id: Math.random().toPrecision(2).toString(),
    title: `Item ${index + 1}`,
  };
};

const getItemCount = () => {
  return 200;
};

const ThirdComponent = (props: ThirdComponentProps) => {
  return (
    <SafeAreaView style={styles.container}>
      <VirtualizedList
        data={DATA}
        initialNumToRender={4}
        renderItem={({ item }) => {
          return (
            <View style={styles.item}>
              <Text style={styles.title}>{item.title}</Text>
              <Button
                title={`Details of ${item.title}`}
                onPress={() =>
                  props.navigation.navigate('ThirdComponentDetails', item)
                }
              />
            </View>
          );
        }}
        keyExtractor={item => item.key}
        getItemCount={getItemCount}
        getItem={getItem}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 20,
  },
  item: {
    display: 'flex',
    backgroundColor: '#f9c2ff',
    flexWrap: 'wrap',
    alignItems: 'flex-start',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginVertical: 8,
    marginHorizontal: 16,
    padding: 20,
  },
  title: {
    fontSize: 32,
  },
});

export default ThirdComponent;
