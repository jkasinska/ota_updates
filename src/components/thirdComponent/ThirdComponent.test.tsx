import { fireEvent, render, RenderAPI } from '@testing-library/react-native';
import 'jest-extended';
import React from 'react';
import { reactNavigationMock } from '../../../__tests__/mocks/reactNavigation.mock';
import ThirdComponent from './ThirdComponent';

describe('ThirdComponent should', () => {
  let component: RenderAPI;
  const props = {
    ...reactNavigationMock,
  };

  beforeEach(() => {
    component = render(<ThirdComponent {...props} />);
  });

  it('match the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('navigate to ThirdComponentDetails', () => {
    const { getByText } = component;

    fireEvent.press(getByText('Details of Item 1'));

    expect(props.navigation.navigate).toHaveBeenCalledTimes(1);
    expect(props.navigation.navigate).toHaveBeenCalledWith(
      'ThirdComponentDetails123',
      {
        id: expect.toBeString(),
        title: 'Item 1',
        key: '0',
      },
    );
  });
});
