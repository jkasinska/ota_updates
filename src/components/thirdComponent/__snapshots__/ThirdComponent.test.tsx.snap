// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThirdComponent should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "marginTop": 20,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={4}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#f9c2ff",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 32,
              }
            }
          >
            Item 1
          </Text>
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Details of Item 1
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#f9c2ff",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 32,
              }
            }
          >
            Item 2
          </Text>
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Details of Item 2
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#f9c2ff",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 32,
              }
            }
          >
            Item 3
          </Text>
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Details of Item 3
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#f9c2ff",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 32,
              }
            }
          >
            Item 4
          </Text>
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Details of Item 4
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
